File: PaymentManagement/src/main/java/com/example/paymentmanagement/service/PaymentService.java
package com.example.paymentmanagement.service;

import com.example.paymentmanagement.dto.PaymentDto;
import com.example.paymentmanagement.model.Method;
import com.example.paymentmanagement.model.Payment;
import com.example.paymentmanagement.repository.PaymentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.Date;
import java.util.List;
import java.util.UUID;
import java.util.stream.Collectors;

@Service
public class PaymentService {

    @Autowired
    private PaymentRepository paymentRepository;

    // Create payment
    public PaymentDto createPayment(PaymentDto dto) {
    Payment payment = Payment.builder()
            
            .paymentDate(new Date())
            .amount(dto.getAmount())
            .paymentMethod(dto.getPaymentMethod())
            .reservationID(dto.getReservationID())
            .userID(dto.getUserID())
            .build();

    return toDto(paymentRepository.save(payment));
}

    // Get payment by ID
    public PaymentDto getPaymentById(String id) {
        Payment payment = paymentRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Payment not found with id: " + id));
        return toDto(payment);
    }

    // Get all payments
    public List<PaymentDto> getAllPayments() {
        return paymentRepository.findAll()
                .stream().map(this::toDto)
                .collect(Collectors.toList());
    }

    // Get payments by user
    public List<PaymentDto> getPaymentsByUser(Integer userID) {
        return paymentRepository.findByUserID(userID)
                .stream().map(this::toDto)
                .collect(Collectors.toList());
    }

    // Get payments by reservation
    public List<PaymentDto> getPaymentsByReservation(Integer reservationID) {
        return paymentRepository.findByReservationID(reservationID)
                .stream().map(this::toDto)
                .collect(Collectors.toList());
    }

    // Get payments by method
    public List<PaymentDto> getPaymentsByMethod(Method method) {
        return paymentRepository.findByPaymentMethod(method)
                .stream().map(this::toDto)
                .collect(Collectors.toList());
    }

    // Delete payment
    public void deletePayment(String id) {
        if (!paymentRepository.existsById(id)) {
            throw new RuntimeException("Payment not found with id: " + id);
        }
        paymentRepository.deleteById(id);
    }

    // Convert to DTO
    private PaymentDto toDto(Payment payment) {
        return PaymentDto.builder()
                .paymentID(payment.getPaymentID())
                .paymentDate(payment.getPaymentDate())
                .amount(payment.getAmount())
                .paymentMethod(payment.getPaymentMethod())
                .reservationID(payment.getReservationID())
                .userID(payment.getUserID())
                .build();
    }
}

File: frontend/src/app/admin/services/payment.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { PaymentDto, PaymentMethod } from '../models/payment.model';
import {environment} from '../../../environments/environment';

const BASE = environment.ApiUrl;

@Injectable({ providedIn: 'root' })
export class PaymentService {
  constructor(private http: HttpClient) {}

  /** POST /api/payments */
  createPayment(dto: PaymentDto): Observable<PaymentDto> {
    return this.http.post<PaymentDto>(`${BASE}/api/payments`, dto);
  }

  /** GET /api/payments */
  getAllPayments(): Observable<PaymentDto[]> {
    return this.http.get<PaymentDto[]>(`${BASE}/api/payments`);
  }

  /** GET /api/payments/{id} */
  getPaymentById(id: string): Observable<PaymentDto> {
    return this.http.get<PaymentDto>(`${BASE}/api/payments/${id}`);
  }

  /** GET /api/payments/user/{userID} */
  getPaymentsByUser(userID: number): Observable<PaymentDto[]> {
    return this.http.get<PaymentDto[]>(`${BASE}/api/payments/user/${userID}`);
  }

  /** GET /api/payments/reservation/{reservationID} */
  getPaymentsByReservation(reservationID: number): Observable<PaymentDto[]> {
    return this.http.get<PaymentDto[]>(`${BASE}/api/payments/reservation/${reservationID}`);
  }

  /** GET /api/payments/method/{method} */
  getPaymentsByMethod(method: PaymentMethod): Observable<PaymentDto[]> {
    return this.http.get<PaymentDto[]>(`${BASE}/api/payments/method/${method}`);
  }

  /** DELETE /api/payments/{id} */
  deletePayment(id: string): Observable<void> {
    return this.http.delete<void>(`${BASE}/api/payments/${id}`);
  }
}
